app/services/search/query_builders/chat_channel_membership.rb

```diff
@@ -70,7 +74,14 @@ def query_conditions
         QUERY_KEYS.map do |query_key|
           next if @params[query_key].blank?
 
-          { match: { query_key => { query: @params[query_key] } } }
+          {
+            simple_query_string: {
+              query: "#{@params[query_key]}*",
```

Comment:
Here we need the ability to search using a wildcard which is why I switched from `match` to [`simple_query_string`](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis.html). Simple query string allows us to search using a wildcard. Now I'm sure you are wondering why not use `query_string` like we do with tags. Well TBH, I just learned that simple_query_string exists and one of the advantages is that...

> While its syntax is more limited than the query_string query, the simple_query_string query does not return errors for invalid syntax. Instead, it ignores any invalid parts of the query string.

We don't need a lot of fancy syntax for just searching a single field so this is the perfect place to use it. I will probably update tag to use it as well since those queries are also simple.

In the future, ideally we will want to set up an [analyzer](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis.html) that will allow us to analyze these fields in such a way that we never need a wildcard to search but that is a much more in-depth process. This will get us what we need for now and will be plenty performant for a while.

Reply:
> We don't need a lot of fancy syntax for just searching a single field so this is the perfect place to use it. I will probably update tag to use it as well since those queries are also simple.

That makes sense, I can see where the syntax is a little simpler. To your first point of:
> Here we need the ability to search using a wildcard which is why I switched from match to simple_query_string

What if we used the [wildcard query](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-wildcard-query.html) instead. It seems a little more verbose, but maybe it's more consistent and avoids playing with 2 different syntaxes in the codebase for Elasticsearch queries? It would also give us access to `boost` and `rewrite` parameters if we need them in the future.

Reply:
One of the big reasons I like the simple query is bc it ignores invalid parts of the query, I believe wildcard would throw an error. I am also not sure if wildcard works with phrases or multiple words, where as the query string will break down multiple words and search them as tokens. I am inclined to slick with this for now and then explore the wildcard option in the future. I would also be interested in performance testing them so see how they do.